var/spack/repos/builtin/packages/libjpeg-turbo/package.py

```diff
@@ -72,3 +77,21 @@ def install(self, spec, prefix):
             cmake('..', *cmake_args)
             make()
             make('install')
+
+    def configure_args(self):
+        args = []
+        if '+java' in self.spec:
+            args.append('--with-java')
+        else:
+            args.append('--without-java')
+        return args
+
+    def setup_environment(self, spack_env, run_env):
+        if '+java' in self.spec:
+            spack_env.set(
+                'JNI_CFLAGS',
+                '-I' + self.spec['jdk'].prefix.include +
+                ' ' + '-I' +
+                os.path.join(self.spec['jdk'].prefix.include,
+                             platform.system().lower()),
```

Comment:
It might be better to handle this in the specific Java package and then use `spec['java'].headers.include_flags`